### PR TITLE
feat(html): add forms helper

### PR DESCRIPTION
**Pull request by epetrow**
part of: https://github.com/telerik/kendo-themes-private/issues/18

**_Note:_** The changes in the Fluent theme are expected and are related to the use of different classes on the label in the tests e.g. `k-label` and `k-form-label`

### CODE REVIEW

**Review by TeyaVes — APPROVED** [review]

File: packages/html/src/form/form-field.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+export const FORMFIELD_CLASSNAME = 'k-form-field';
+
+export type KendoFormFieldProps = {
+    label?: null | string;
+    optional?: boolean;
+    component?: JSX.Element | string;
+    hint?: null | string;
+    error?: null | string;
+    info?: null | string;
+    disabled?: boolean;
```
Comment (TeyaVes) [inline packages/html/src/form/form-field.tsx:12]: we can add the disabled prop as a state here

File: tests/grid/grid-editing-popup.html [hunk 2]
```diff
@@ -86,20 +80,18 @@
...
-                            <div class="k-form-label">
-                                <label for="Discontinued">Discontinued</label>
-                            </div>
+                            <label class="k-label k-form-label">Discontinued</label>
                             <div class="k-form-field-wrap">
-                                <span class="k-checkbox-wrap">
-                                    <input id="Discontinued" type="checkbox" class="k-checkbox k-checkbox-md k-rounded-md">
-                                </span>
-                                <label class="k-checkbox-label">​</label>
+                                <div class="k-form-field-checkbox-wrapper">
```
Comment (Juveniel) [inline tests/grid/grid-editing-popup.html:85]: we do not need  the div with class `k-form-field-checkbox-wrapper`

File: tests/scheduler/scheduler-edit-dialog.html [hunk 3]
```diff
@@ -84,30 +78,32 @@
...
                         </div>
                         <div class="k-form-field">
-                            <div class="k-form-label k-empty"></div>
+                            <span class="k-label k-form-label k-label-empty"></span>
                             <div class="k-form-field-wrap">
-                                <span class="k-checkbox-wrap">
-                                    <input id="k-is-allday-chkbox" type="checkbox" class="k-checkbox k-checkbox-md k-rounded-md">
-                                </span>
-                                <label class="k-checkbox-label" for="k-is-allday-chkbox">All Day Event</label>
+                                <div class="k-form-field-checkbox-wrapper">
```
Comment (Juveniel) [inline tests/scheduler/scheduler-edit-dialog.html:83]: `<div class="k-form-field-checkbox-wrapper">` should be removed

File: tests/scheduler/scheduler-edit-dialog.html [hunk 4]
```diff
@@ -84,30 +78,32 @@
...
                         </div>
                         <div class="k-form-field">
-                            <div class="k-form-label k-empty"></div>
+                            <span class="k-label k-form-label k-label-empty"></span>
                             <div class="k-form-field-wrap">
-                                <span class="k-checkbox-wrap">
-                                    <input id="k-set-timezone" type="checkbox" class="k-checkbox k-checkbox-md k-rounded-md">
-                                </span>
-                                <label class="k-checkbox-label" for="k-set-timezone">Specify Time Zone</label>
+                                <div class="k-form-field-checkbox-wrapper">
```
Comment (Juveniel) [inline tests/scheduler/scheduler-edit-dialog.html:94]: `<div class="k-form-field-checkbox-wrapper">` should be removed

File: packages/html/src/form/form-field.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    alignText?: 'start' | 'end';
```
Comment (Juveniel) [inline packages/html/src/form/form-field.tsx:13]: is `alignText` being used anywhere ? If not, please remove it.

File: packages/html/src/form/form-field.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+            {emptyLabel && <span className="k-label k-form-label k-label-empty"></span>}
+            <ConditionalWrapper
+                condition={fieldWrap}
+                wrapper={children => <div className="k-form-field-wrap">{children}</div>}>
+                <ConditionalWrapper
+                    condition={checkboxLabel}
+                    wrapper={children => <div className="k-form-field-checkbox-wrapper">{children}</div>}>
+                    {component}
+                    {checkboxLabel && <label className="k-checkbox-label">{checkboxLabel}</label>}
+                </ConditionalWrapper>
```
Comment (Juveniel) [inline packages/html/src/form/form-field.tsx:71]: I am not sure we need those conditionals as:

1) The `k-form-field-wrap` can be rendered everytime
2) The checkbox conditional is not needed, as the `<Checkbox />` spec already adds its required wrapper.

**Review by Juveniel — CHANGES_REQUESTED** [review]

**Pull request merged**